vscode-lean4/src/utils/leanpkg.ts

```diff
@@ -67,11 +68,11 @@ export class LeanpkgService implements Disposable {
         // Note: just opening the file fires this event sometimes which is annoying, so
         // we compare the contents just to be sure and normalize whitespace so that
         // just adding a new line doesn't trigger the prompt.
-        const [workspaceFolder, packageUri, packageFileUri] = findLeanPackageRoot(uri);
+        const [workspaceFolder, packageUri, packageFileUri] = await findLeanPackageRoot(uri);
         if (packageUri) {
-            const fileUri = this.findLakeFile(packageUri);
+            const fileUri = await this.findLakeFile(packageUri);
             if (fileUri) {
-                const contents = this.readWhitespaceNormalized(fileUri);
```

Comment:
this can throw which means it would be an unhandled exception in this function so handleLakeFileChanged needs a try catch.  I think the right semantic if an exception is thrown would be to pretend the file no longer exists so the `existing` content should go to empty string and the `this.lakeFileChangedEmitter.fire` even should be raised.